libbeat/publisher/queue/memqueue/queue_test.go

```diff
@@ -74,6 +76,64 @@ func TestProduceConsumer(t *testing.T) {
 	t.Run("flush", testWith(makeTestQueue(bufferSize, batchSize/2, 100*time.Millisecond)))
 }
 
+func TestQueueMetrics(t *testing.T) {
+	eventsToTest := 5
+	maxEvents := 10
+
+	// Test the directEventLoop
+	directSettings := Settings{
+		Events:         maxEvents,
+		FlushMinEvents: 1,
+		FlushTimeout:   0,
+	}
+	t.Logf("Testing directEventLoop")
+	queueTestWithSettings(t, directSettings, eventsToTest, "directEventLoop")
+
+	// Test Buffered Event Loop
+	bufferedSettings := Settings{
+		Events:         maxEvents,
+		FlushMinEvents: eventsToTest, // The buffered event loop can only return FlushMinEvents per Get()
+		FlushTimeout:   time.Millisecond,
+	}
+	t.Logf("Testing bufferedEventLoop")
+	queueTestWithSettings(t, bufferedSettings, eventsToTest, "bufferedEventLoop")
```

Comment:
The separate cases here could probably be sub-tests so they can be run individually.